File: frontend/src/pages/dashboard/projects/types/project.ts
export interface Project {
  id: number;
  name: string;
  description: string | null;
  company_id: number;
  department_id: number;
  start_date: string;
  end_date: string | null;
  status: string;
  budget: number;
  created_at: string;
}

export interface ProjectCreateRequest {
  name: string;
  description?: string | null;
  company_id: number;
  department_id: number;
  start_date: string;
  end_date?: string | null;
  status: string;
  budget: number;
}

export interface ProjectUpdateRequest {
  name?: string;
  description?: string | null;
  company_id?: number;
  department_id?: number;
  start_date?: string;
  end_date?: string | null;
  status?: string;
  budget?: number;
}

File: backend/app/models/project.py
from sqlalchemy import (
    Column,
    Integer,
    String,
    Date,
    Numeric,
    DateTime,
    ForeignKey,
)
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.database.database import Base


class Project(Base):
    __tablename__ = "projects"

    id = Column(
        Integer,
        primary_key=True,
        index=True,
    )

    name = Column(
        String(150),
        nullable=False,
    )

    description = Column(
        String(500),
        nullable=True,
    )

    company_id = Column(
        Integer,
        ForeignKey("companies.id", ondelete="CASCADE"),
        nullable=False,
    )

    department_id = Column(
        Integer,
        ForeignKey("departments.id", ondelete="CASCADE"),
        nullable=False,
    )

    start_date = Column(
        Date,
        nullable=False,
    )

    end_date = Column(
        Date,
        nullable=True,
    )

    status = Column(
        String(30),
        default="PLANNED",
        nullable=False,
    )

    budget = Column(
        Numeric(12, 2),
        nullable=False,
    )

    created_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
    )

    company = relationship(
        "Company",
        back_populates="projects",
    )

    department = relationship(
        "Department",
        back_populates="projects",
    )

    tasks = relationship(
    "Task",
    back_populates="project",  
    )